### PR TITLE
Profiles: additional records sheet

### PR DESCRIPTION
**Pull request by estebanmino**
Fixes RNBW-2613

## What changed (plus any additional context for devs)

Added additional records sheet

I had to add `fast-text-encoding` polyfill to be able to validate content hash records, TextEncoder and needed to add it 

## PoW (screenshots / screen recordings)

https://user-images.githubusercontent.com/12115171/162288215-5d61c1a0-c6c6-40d1-b835-3a4bd1524aea.mp4

## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels?
- [ ] Added e2e tests? if not please specify why
- [ ] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

**Comment by benisgold** [comment]
Looks like somethings up with the Twitter placeholder 

Screen Shot 2022-04-07 at 4 29 38 PM

**Comment by benisgold** [comment]
I think we should standardize the separator between keywords, i.e. if the user tries to type a space or slash or whatever between each keyword it should be automatically replaced with ", " (or some other separator) as they type.

**Comment by estebanmino** [comment]
@benisgold yeah totally agree, we need to get this one done for that https://linear.app/rainbow/issue/RNBW-3099/use-text-input-mask-in-records-form so users won't be able to type a space if we don't want them to

File: src/hooks/useENSRegistrationForm.ts [hunk 1]
```diff
@@ -205,13 +205,25 @@ export default function useENSRegistrationForm({
...
+            if (regex && value && !value.match(regex)) {
+              return {
+                ...currentErrors,
+                [key]: message,
+              };
+            }
+          }
+          if (validations?.onSubmit?.validate) {
+            const { callback, message } = validations?.onSubmit?.validate || {};
+            if (value && !callback(value)) {
```
Comment (jxom) [inline src/hooks/useENSRegistrationForm.ts:221]: love this

File: src/languages/_english.json [hunk 2]
```diff
@@ -610,7 +610,33 @@
...
+        "eth": "Ethereum",
+        "btc": "Bitcoin",
+        "ltc": "Litecoin",
+        "doge": "Doge",
+        "wallet_placeholder": "Add a %{coin} address",
+        "content": "Content",
+        "content_placeholder": "Add a content hash",
+        "notice": "Notice",
+        "notice_placeholder": "Add a notice",
+        "keyworkds": "Keywords",
```
Comment (jxom) [inline src/languages/_english.json:623]: typo

File: src/helpers/ens.ts [hunk 3]
```diff
@@ -94,6 +99,7 @@ export type TextRecordField = {
 
 export const textRecordFields = {
   [ENS_RECORDS.displayName]: {
+    defaultRecord: true,
```
Comment (jxom) [inline src/helpers/ens.ts:102]: Wonder if this naming could get confused with having that record as a default selected field (and not a default field on the sheet). 

Maybe better naming could be as simple/direct as `isPrimaryDisplayRecord`?

**Comment by jxom** [comment]
Bottom padding seems a bit off for iPhone 13 Pro, can we add some padding?:

Screen Shot 2022-04-08 at 1 08 06 pm

When I select "Pronouns" and then dismiss the sheet, I think I would expect to see it selected here?

Screen Shot 2022-04-08 at 1 08 00 pm

File: src/languages/_english.json (hunk 2 above)
Comment (estebanmino) [reply src/languages/_english.json:623]: 🤦

File: src/helpers/ens.ts (hunk 3 above)
Comment (estebanmino) [reply src/helpers/ens.ts:102]: cool i'll change it

**Comment by estebanmino** [comment]
@jxom regarding the second comment, nice suggestion! added it https://recordit.co/hAtPKB5DeB

**Comment by estebanmino** [comment]
for the padding comment,
image
since the sheet height is static and not dependent in the content height i added a method that will add additional height depending on the device width

**Review by benisgold — APPROVED** [review]

**Review by jxom — APPROVED** [review]

**Pull request merged**